src/SonarQube.MSBuild.Tasks/WriteProjectInfoFile.cs

```diff
@@ -331,6 +320,32 @@ private bool TryGetSettingValue(ITaskItem taskItem, out string metadataValue)
             return success;
         }
 
+        internal /* for testing purpose */ string GetProjectGuid()
+        {
+            if (!string.IsNullOrEmpty(ProjectGuid))
+            {
+                return ProjectGuid;
+            }
+
+            if (!string.IsNullOrEmpty(SolutionConfigurationContents))
+            {
+                var fullProject = new FileInfo(FullProjectPath);
```

Comment:
I wasn't sure whether `FullProjectPath` was always correct so I am wrapping it up.